api/candidatesApiClient.ts

```diff
@@ -20,11 +21,23 @@ export async function getCandidates(page: number, pageSize: number): Promise<Lis
     const apiURL = publicRuntimeConfig.API_URL;
     const response = await fetch(`${apiURL}/candidates?page=${page}&pageSize=${pageSize}`);
 
-    if(response.ok){
+    if (response.ok) {
         return await response.json();
-    }
-    else
-    {
+    } else {
         throw Error;
     }
-}
\ No newline at end of file
+}
+
+export async function getCandidateById(cid: any): Promise<Candidate> {
+    const { publicRuntimeConfig } = getConfig();
+    const apiURL = publicRuntimeConfig.API_URL;
+    const response = await fetch(`${apiURL}/candidates`);
+    if (response.ok) {
+        const data = await response.json();
+        const candidate = data.items.find(item => item.id === parseInt(cid));
+        return candidate;
```

Comment:
we do have a `/candidates/{candidateID}` endpoint, which is what we should use here.
(although I'll give you that it's returning a 500 response right now...)

This will fail as soon as we have more than 10 candidates, because the candidate we are looking for won't always be in the first page of results.

I'll raise a ticket for fixing the issue with that API endpoint!

Reply:
thanks!

Reply:
this is now updated and seems to work as long as the id exists